### PR TITLE
Added Pop ups for Login/Registration Errors and Order Confirmation

### LINKED ISSUE #85: Convert messages to popup windows
convert existing error messages to popups:
- login error
- register error
- no matching searches
- etc

maybe also add success messages as popups:
- new user created
- item added/removed
- checkout completed (maybe)

### PR DESCRIPTION
**Pull request by maishabd23**
Screenshot (91)
Incorrect Username pop up

Screenshot (92)
Invalid password

Screenshot (89)
When user tries to register with a non unique username

Screenshot (90)
Order Completed pop up

### CODE REVIEW

File: src/main/java/bookstore/inventory/CheckoutController.java [hunk 1]
```diff
@@ -387,6 +387,7 @@ public String confirmOrder(HttpServletRequest request, HttpServletResponse respo
         // Generate a random confirmation number
         String confirmationNumber = UUID.randomUUID().toString();
         model.addAttribute("confirmationNumber", confirmationNumber);
+        model.addAttribute("confirmationMessage", "Order Completed!");
```
Comment (thanujasiva) [inline src/main/java/bookstore/inventory/CheckoutController.java:390]: Hmm, with this pop-up, do we still want it to go to the order confirmation page? Could the confirmation number be added to the pop up maybe?

Does it make sense to go to the confirmation page, or go back to listAvailableBooks?

**Review by thanujasiva — CHANGES_REQUESTED** [review]

File: src/main/java/bookstore/users/UserController.java [hunk 2]
```diff
@@ -113,23 +113,23 @@ public String createAccountSubmit(@ModelAttribute BookUser bookUser,
             List<BookUser> existingUsers = userRepository.findByUsername(bookUser.getUsername());
 
             if (!existingUsers.isEmpty()) { // only allow accounts with unique usernames
-                model.addAttribute("errorType", "registration");
-                model.addAttribute("error", "Username already exists. Please use a new username or login with the current username.");
-                return "accountError";
```
Comment (thanujasiva) [inline src/main/java/bookstore/users/UserController.java:118]: If `accountError.html` isn't being used anymore, can it be removed?

**Review by thanujasiva — CHANGES_REQUESTED** [review]

File: src/main/java/bookstore/inventory/CheckoutController.java (hunk 1 above)
Comment (maishabd23) [reply src/main/java/bookstore/inventory/CheckoutController.java:390]: I was thinking the order complete confirmation number might be too long for the pop-up and the pop up fades after a while. Do we want the confirmation number to fade away?

File: src/main/java/bookstore/users/UserController.java (hunk 2 above)
Comment (maishabd23) [reply src/main/java/bookstore/users/UserController.java:118]: done

**Review by thanujasiva — APPROVED** [review]

File: src/main/java/bookstore/inventory/CheckoutController.java (hunk 1 above)
Comment (thanujasiva) [reply src/main/java/bookstore/inventory/CheckoutController.java:390]: Yeah that makes sense, we can keep the confirmation as is

**Review by sabahsamwatin — APPROVED** [review]

**Pull request merged**